exp/ipam/api/v1alpha1/conversion.go

```diff
@@ -49,13 +50,42 @@ func (dst *IPAddressList) ConvertFrom(srcRaw conversion.Hub) error {
 func (src *IPAddressClaim) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*ipamv1.IPAddressClaim)
 
-	return Convert_v1alpha1_IPAddressClaim_To_v1beta1_IPAddressClaim(src, dst, nil)
+	if err := Convert_v1alpha1_IPAddressClaim_To_v1beta1_IPAddressClaim(src, dst, nil); err != nil {
+		return err
+	}
+
+	if src.ObjectMeta.Labels != nil {
+		dst.Spec.ClusterName = src.ObjectMeta.Labels["cluster.x-k8s.io/cluster-name"]
+		if dst.ObjectMeta.Annotations != nil && dst.ObjectMeta.Annotations["conversion.cluster.x-k8s.io/cluster-name-label-set"] != "" {
+			delete(src.ObjectMeta.Labels, "cluster.x-k8s.io/cluster-name")
```

Comment:
Why are we deleting something in the src object? What impact does it have to delete something in the src object during conversion?

Reply:
I think this is a typo and it should really be `dst` here. But since the fuzzy tests seem to call this method and and it has a pointer receiver, removing it on the source object also 'fixes' the tests. I'll change it, unless we decide for the common  json serialisation approach.